themes/openy_themes/openy_carnation/src/js/openy_carnation.js

```diff
@@ -0,0 +1,460 @@
+        $('body').addClass('with-banner');
+      } else {
+        $('body').addClass('without-banner');
+      }
+    }
+  };
+
+  /**
+   * Ensure breadcrumbs are after banners in the DOM
+   */
+  Drupal.behaviors.moveBreadcrumbs = {
+    attach: function (context, settings) {
+      var breadCrumbs = $('.breadcrumbs-wrapper', context);
+      var bannerCta = $(
+        '.banner-zone-node .banner .banner-cta, ' +
+        '.banner-zone-node .landing-header, ' +
+        '.banner-zone-node .paragraph-gallery', context);
+
+      if (breadCrumbs.length && bannerCta.length) {
+        breadCrumbs.once('openy-breadcrumbs').appendTo(bannerCta);
+      }
+    }
+  };
+
+  /**
+   * Ensure header alerts are after breadcrumbs in the DOM
+   */
+  Drupal.behaviors.moveHeaderAlerts = {
+    attach: function (context, settings) {
+      var headerAlerts = $('#block-openy-carnation-views-block-alerts-header-alerts', context);
+      var subHeaderFilters = $('.banner-cta', context);
+
+      if (headerAlerts.length && subHeaderFilters.length) {
+        headerAlerts.once('openy-breadcrumbs').insertAfter(subHeaderFilters);
+      }
+    }
+  };
+
+  /**
+   * Show/hide desktop search block.
+   */
+  Drupal.behaviors.openySearchToggle = {
+    attach: function (context, settings) {
+      var searchBtn = $('.site-search button');
+      var mainMenuLinks = $('.page-head__main-menu .nav-level-1 li:not(:eq(0))').find('a, button');
+      var searchClose = $('.page-head__search-close');
+
+      searchBtn.once('openy-search-toggle-hide').on('click', function () {
+        mainMenuLinks.removeClass('show').addClass('fade');
+      });
+
+      searchClose.once('openy-search-toggle-show').on('click', function () {
+        mainMenuLinks.addClass('show');
+      });
+    }
+  };
+
+  /**
+   * Add class to header when mobile menu is opened.
+   */
+  Drupal.behaviors.openyMobileMenu = {
+    attach: function (context, settings) {
+      var sidebar = $('#sidebar');
+      var $target = $('.top-navs');
+
+      sidebar.on('show.bs.collapse', function () {
+        $target.addClass('menu-in');
+      });
+
+      sidebar.on('hide.bs.collapse', function () {
+        $target.removeClass('menu-in');
+      });
+    }
+  };
+
+  /**
+   * Dropdown menu height.
+   */
+  Drupal.behaviors.openyDropdownMenu = {
+    attach: function (context, settings) {
+      $('.nav-desktop .nav-level-2').each(function (index, element) {
+        var item = $(element);
+        var offset = item.offset();
+        var bottom = offset.top + item.height() - $(window).scrollTop();
+        var maxHeight = $(window).height() - bottom - 15;
+        item.find('.dropdown-menu').eq(0).css('max-height', maxHeight);
+      });
+    }
+  };
+
+  /**
+   * BS4 data-spy: affix replacement
+   */
+  Drupal.behaviors.openyHeaderAffix = {
+    attach: function (context, settings) {
+      $(window).on('scroll', function(event) {
```

Comment:
probably we need to use once() function here. Lets discuss what this behavior is doing. Not sure about it.

Reply:
This adds a class to the header so that it turns from transparent to a solid colour after scrolling. I've added the .once() to the function though! Affix was removed in bootstrap 4.